src/google_form_to_engagement_db/google_form_to_engagement_db.py

```diff
@@ -290,15 +311,42 @@ def _sync_google_form_to_engagement_db(google_form_client, engagement_db, form_c
             sync_stats.add_event(GoogleFormSyncEvents.READ_ANSWER_FROM_RESPONSE)
             if answer["questionId"] not in question_id_to_engagement_db_dataset:
                 log.info(f"This answer is to question {answer['questionId']}, which isn't configured in this sync")
-                continue
+                continue 
 
-            message = _form_answer_to_engagement_db_message(
+            question_id_to_engagement_db_message[answer["questionId"]] = _form_answer_to_engagement_db_message(
                 answer, form_config.form_id, response, participant_uuid, question_id_to_engagement_db_dataset
             )
-            message_origin_details = {
+            question_id_to_message_origin_details[answer["questionId"]] = {
                 "formId": form_config.form_id,
                 "answer": answer,
             }
+
+        for question_config in form_config.question_configurations:
+            assert len(question_config.question_titles) > 0
+            if len(question_config.question_titles) == 1:
+                question_id = question_title_to_question_id[question_config.question_titles[0]]
+                if question_id not in question_id_to_engagement_db_message:
+                    continue
+                message = question_id_to_engagement_db_message[question_id]
+                message_origin_details = question_id_to_message_origin_details[question_id]
```

Comment:
Do you need this branch or is it taken care of by the if len(messages) == 1 case below?

Reply:
Yes, it's needed. The case `len(messages) == 1` enables us not to merge messages in case a participant answers only one of the multiple questions given.